numpy/core/src/multiarray/_datetime.h

```diff
@@ -287,8 +286,7 @@ convert_timedelta_to_pyobject(npy_timedelta td, PyArray_DatetimeMetaData *meta);
 /*
  * Converts a datetime based on the given metadata into a datetimestruct
  */
-NPY_NO_EXPORT int
-convert_datetime_to_datetimestruct(PyArray_DatetimeMetaData *meta,
+int convert_datetime_to_datetimestruct(PyArray_DatetimeMetaData *meta,
```

Comment:
I don't think this is sufficient, extension modules aren't allowed to link against each other. This needs to go through the function table like the other `NUMPY_API` functions.

Reply:
Thanks, ill see if i can track that down.  Before I do, is this likely to be accepted once working?

Reply:
I think we'd need to think about the function names, but in principle it seems sensible to me to expose the `npy_datetimestruct` type and some APIs that consume it.